contracts/src/interfaces/IPayerReport.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+/**
+ * @title IPayerReport
+ * @notice Interface for the PayerReport contract handling usage reports and batch settlements.
+ */
+interface IPayerReport {
+    //==============================================================
+    //                             STRUCTS
+    //==============================================================
+
+    struct UsageReport {
+        address originatorNode;
+        uint256 startingSequenceID;
+        uint256 endingSequenceID;
+        uint256 lastMessageTimestamp;
+        uint256 reportTimestamp;
+        bytes32 reportMerkleRoot;
+        address[] payers;
+        uint256[] amountsSpent;
+    }
+
+    //==============================================================
+    //                          EVENTS
+    //==============================================================
+
+    /**
+     * @dev Emitted when an originator node submits a usage report.
+     * The report includes the Merkle root of the detailed (payer, amount) data.
+     * Note: Payers and amounts are not stored on-chain, only emitted in the event.
+     * Nodes listen to this event to get full details of the report.
+     */
+    event PayerReportSubmitted(
+        address indexed originatorNode,
+        uint256 indexed reportIndex,
+        bytes32 indexed reportMerkleRoot,
+        uint256 startingSequenceID,
+        uint256 endingSequenceID,
+        uint256 lastMessageTimestamp,
+        uint256 reportTimestamp,
+        address[] payers,
+        uint256[] amountsSpent
+    );
+
+    /**
+     * @dev Emitted when a node attests to the correctness of a report.
+     */
+    event PayerReportAttested(
+        address indexed originatorNode,
+        uint256 indexed reportIndex,
+        bytes32 indexed reportMerkleRoot
+    );
+
+    /**
+     * @dev Emitted when a usage report is confirmed.
+     */
+    event PayerReportConfirmed(
+        address indexed originatorNode,
+        uint256 indexed reportIndex,
+        bytes32 indexed reportMerkleRoot
+    );
+
+    /**
+     * @dev Emitted when a batch of usage is settled.
+     */
+    event UsageSettled(
+        address indexed originatorNode,
+        uint256 indexed reportIndex,
+        uint256 offset,
```

Comment:
For this event to be indexable by the nodes I believe we would need to have the payers and the amounts

Reply:
Otherwise the node would have to do a bunch of work to reconstruct the tree and figure out what was being settled.

Reply:
Agree, modifying this part too.

Also, with the latest commit I added a `uint16 batches` to the PayerReport struct, which should be calculated off-chain based on `getMaxBatchSize()`. Other nodes can reconstruct that data and agree on it.

I think this is interesting because we don't submit the amount of payers and amounts anymore, so I don't see a reliable way of calculating the amount of batches expected for a single report, tracking what was already settled, etc.

Reply:
This makes sense to me. Do we want it to be the number of batches, or just the number of leaves in the tree? I'm fine with either, but the number of leaves lets the caller adjust the batch size as needed.

Reply:
I like the number to be leaves in the tree. It reduces ambiguity and hidden knowledge (such as batches has to be N payers max)